etc/ansible/roles/network-runner/providers/dellemc.os9.os9/conf_access_port.yaml

```diff
@@ -9,13 +9,60 @@
   ansible.builtin.set_fact:
     port_description: "{{ port_description_cmd.stdout[0] | regex_search('(?<=description ).+?(?=[\\n\\r])') }}"
 
+- name: "dellemc.os9: get existing vlans from running config"
+  dellemc.os9.os9_command:
+    commands: "show interfaces switchport {{ port_name }}"
+  register: port_vlans_cmd
+  connection: network_cli
+
+- name: "dellemc.os9: parse untagged vlan and tagged vlans from running config"
+  ansible.builtin.set_fact:
+    port_has_untagged: false
+    port_untagged: "{{ port_vlans_cmd.stdout[0] | regex_search('U\\s+(\\d+(?=[\\n\\r]|$))', '\\1') }}"
+    port_has_tagged: false
+    port_tagged: "{{ port_vlans_cmd.stdout[0] | regex_search('T\\s+([\\d,-]+(?=[\\n\\r]|$))', '\\1') }}"
+
+- name: "dellemc.os9: check if untagged vlan was found"
+  ansible.builtin.set_fact:
+    port_untagged_str: "{{ port_untagged[0] }}"
+    port_has_untagged: true
+  when: ( port_untagged is iterable ) and ( port_untagged is not string ) and ( port_untagged | length > 0 ) and ( port_untagged[0] != "1" )
+
+- name: "dellemc.os9: check if tagged vlans were found"
+  ansible.builtin.set_fact:
+    port_tagged_str: "{{ port_tagged[0] }}"
+    port_has_tagged: true
+    port_tagged_list: []
+  when: ( port_tagged is iterable ) and ( port_tagged is not string ) and ( port_tagged | length > 0 )
+
+- name: "dellemc.os9: parse ranges for tagged vlans"
+  ansible.builtin.set_fact:
+    port_tagged_list: "{{ port_tagged_list + (range(item.split('-')[0] | int, (item.split('-')[1] | int + 1)) | list if '-' in item else [item]) | map('string') }}"
+  loop: "{{ port_tagged_str.split(',') }}"
+  when: port_has_tagged
+
+- name: "dellemc.os9: Remove untagged vlan from port"
+  dellemc.os9.os9_config:
+    lines:
+      - "no untagged {{ port_name }}"
+    parents: ["interface Vlan {{ port_untagged_str }}"]
+  connection: network_cli
+  when: port_has_untagged
+
+- name: "dellemc.os9: Remove tagged vlans from port"
+  dellemc.os9.os9_config:
+    lines:
+      - "no tagged {{ port_name }}"
+    parents: ["interface Vlan {{ item }}"]
+  loop: "{{ port_tagged_list }}"
+  connection: network_cli
+  when: port_has_tagged
```

Comment:
Yeah, this makes sense especially since we have to put this in 2 files. @tzumainn @naved001 do you know if I can just put a filter plugin in the `filter plugins` folder and it will just be found? I was mostly trying to keep it in 1 file because I don't really know how these tasks are invoked in neutron. I guess what I'm asking is what is the playbook that gets run in this repo that references these tasks?

Reply:
@hakasapl I am not sure really, you can try to invoke network runner code to see if things work correctly. I bet this is how neutron/ml2 is trying to invoke the networking change.

```python
from network_runner.models.inventory import Inventory
from network_runner.models.inventory import Host

hostname = 'MOC-R4PAC21-SW-TORS-A'
host = Host(name=hostname,
            ansible_host='10.2.10.1',
            ansible_user='admin',
            ansible_become='true',
            ansible_become_method='enable',
            ansible_become_password='password',
            ansible_ssh_pass='password',
            ansible_network_os='dellemc.os9.os9')

inventory = Inventory()
inventory.hosts.add(host)

vlan = 252
vlan_name = 'MOC Research Ceph'
port = 'twentyfivegige 1/7/4'

trunk_vlans = [259, 250]

from network_runner import api
netrunner = api.NetworkRunner(inventory)

netrunner.conf_access_port(hostname, port, vlan)
```